### PR TITLE
infoschema: Migrate the infoschema's retrieving data logic for 'CharacterSets/Collations' to executor 

### PR DESCRIPTION
**Pull request by AndrewDi**
### What problem does this PR solve? 
Infoschema contains too much logic for retrieving data, which is confusing, we need to migrate this part to the executor so that infoshema only contains schema information. Detail ref #15040 #15035 

### What is changed and how it works?
migrate function dataForCharacterSets/dataForCollations from infoschema pkg to executor/infoschema_reader.go

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

### CODE REVIEW

**Comment by AndrewDi** [comment]
@lonng @crazycs520 PTAL

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ package executor
 
 import (
 	"context"
+	"github.com/pingcap/parser/charset"
```
Comment (reafans) [inline executor/infoschema_reader.go]: move this line to line-21

Comment (AndrewDi) [reply executor/infoschema_reader.go]: done

**Comment by reafans** [comment]
lgtm

**Pull request merged**